Resolve issues transferring built graph to host

rbreslow opened the pull request:
## Overview

* Switch from using `fetch` to `synchronize`

Fixes #1083 

## Notes

I was able to get the graph transfer to be successful, even using the `fetch` Ansible module. With the larger graph from production, I did notice some slowdown on my machine during the transfer. After switching to using `synchronize`, transfers are faster and the slowdown is gone.

 I think that https://github.com/ansible/ansible/issues/31194 supports a theory that this is related to excessive memory consumption. It's possible that hardware differences allowed transfers to succeed on my machine.

## Testing Instructions

First, I downloaded the `Graph.obj` from the production EC2 instance running in the Clean Air Council AWS account. 

Then, I executed `./scripts/vagrant-up.sh`.

Next, I placed the `Graph.obj` file in the `otp` VM:

```bash
$ scp Graph.obj vagrant@192.168.8.26:/tmp/Graph.obj
$ vagrant ssh otp
vagrant@otp:~$ cd /var/otp
vagrant@otp:/var/otp$ sudo cp /tmp/Graph.obj .
vagrant@otp:/var/otp$ sudo chown root:root Graph.obj
vagrant@otp:/var/otp$ sudo cp Graph.obj default/Graph.obj
```

I removed the `Graph.obj` file created earlier and reprovisioned the `otp` VM to allow the copy task to execute again:

```bash
$ rm otp_data/Graph.obj
$ time vagrant up --provision otp
...
TASK [cac-tripplanner.otp-data : Create Directory for OSM and GTFS] ************
ok: [otp]

TASK [cac-tripplanner.otp-data : Copy Feed Validator] **************************
ok: [otp]

TASK [cac-tripplanner.otp-data : Install AuthBind (for serving OTP on port 80)] ***
ok: [otp]

TASK [cac-tripplanner.otp-data : Touch AuthBind file if it does not exist] *****
changed: [otp]

TASK [cac-tripplanner.otp-data : Change owner and mode of byport file] *********
ok: [otp]

TASK [cac-tripplanner.otp-data : Overwrite upstart service] ********************
changed: [otp]

TASK [cac-tripplanner.otp-data : Create data directory (test)] *****************
ok: [otp]

TASK [cac-tripplanner.otp-data : Download OTP Data (test)] *********************
skipping: [otp]

TASK [cac-tripplanner.otp-data : Copy OTP Data (test/develop)] *****************
ok: [otp]

TASK [cac-tripplanner.otp-data : Check for Existing Graph (develop)] ***********
ok: [otp]

TASK [cac-tripplanner.otp-data : Build OTP Graph (test/develop)] ***************
skipping: [otp]

TASK [cac-tripplanner.otp-data : Copy Built OTP Graph to Host (develop)] *******
changed: [otp]

TASK [cac-tripplanner.otp-data : Create Graph Directory] ***********************
ok: [otp]

TASK [cac-tripplanner.otp-data : Copy Router Configuration to Graph Directory] ***
ok: [otp]

TASK [cac-tripplanner.otp-data : Move Graph to Graph Directory (test/develop)] ***
changed: [otp]

TASK [cac-tripplanner.otp-data : Copy Local Graph to Graph Directory (production)] ***
skipping: [otp]

RUNNING HANDLER [azavea.opentripplanner : Restart OpenTripPlanner] *************
changed: [otp]

PLAY RECAP *********************************************************************
otp                        : ok=34   changed=8    unreachable=0    failed=0


real	1m20.561s
user	0m22.032s
sys	0m7.508s
```

Finally, ensure the transfer was successful:

```bash
$ shasum Graph.obj && ls -lh Graph.obj
a96c7286a23f2c57c6b3195991a160fe140dc20f  Graph.obj
-rw-r--r--  1 rbreslow  AZVA-INT\Domain Users   890M Sep 24 11:22 Graph.obj
```

```bash
vagrant@otp:/var/otp/default$ shasum Graph.obj && ls -lh Graph.obj
a96c7286a23f2c57c6b3195991a160fe140dc20f  Graph.obj
-rw-r--r-- 1 root root 890M Sep 24 15:22 Graph.obj
```

rbreslow commented:
Kathryn, I know you stated in the issue that you tried using this module but experienced a failure. If you experience the same failure when reviewing this PR, could you post your local output of `$ env`?

flibbertigibbet commented:
`env`:

<Details>
CLUTTER_IM_MODULE=xim
NVM_DIR=/home/kat/.nvm
LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:
LESSCLOSE=/usr/bin/lesspipe %s %s
XDG_MENU_PREFIX=gnome-
LANG=en_US.UTF-8
DISPLAY=:0
GNOME_SHELL_SESSION_MODE=ubuntu
NODE_PATH=/home/kat/.npm
GPG_TTY=/dev/pts/2
GTK2_MODULES=overlay-scrollbar
COLORTERM=truecolor
NVM_CD_FLAGS=
USERNAME=kat
JAVA_HOME=/usr/lib/jvm/java-8-oracle
J2SDKDIR=/usr/lib/jvm/java-8-oracle
XDG_VTNR=2
SSH_AUTH_SOCK=/run/user/1001/keyring/ssh
MANDATORY_PATH=/usr/share/gconf/ubuntu.mandatory.path
S_COLORS=auto
XDG_SESSION_ID=2
DERBY_HOME=/usr/lib/jvm/java-8-oracle/db
USER=kat
DESKTOP_SESSION=ubuntu
QT4_IM_MODULE=xim
TEXTDOMAINDIR=/usr/share/locale/
GRADLE_HOME=/opt/gradle/gradle-4.9
GNOME_TERMINAL_SCREEN=/org/gnome/Terminal/screen/11003427_cc60_4645_b4e2_58738b438705
DEFAULTS_PATH=/usr/share/gconf/ubuntu.default.path
QT_QPA_PLATFORMTHEME=appmenu-qt5
PWD=/projects/cac-tripplanner
HOME=/home/kat
TEXTDOMAIN=im-config
J2REDIR=/usr/lib/jvm/java-8-oracle/jre
SSH_AGENT_PID=2671
QT_ACCESSIBILITY=1
LIBVIRT_DEFAULT_URI=qemu:///system
XDG_SESSION_TYPE=x11
XDG_DATA_DIRS=/usr/share/ubuntu:/usr/local/share:/usr/share:/var/lib/snapd/desktop:/var/lib/snapd/desktop
NVM_IOJS_ORG_MIRROR=https://iojs.org/dist
NVM_PATH=/home/kat/.nvm/versions/node/v6.14.4/lib/node
CAC_APP_MEMORY=1024
XDG_SESSION_DESKTOP=ubuntu
CAC_DATABASE_MEMORY=512
GJS_DEBUG_OUTPUT=stderr
GTK_MODULES=gail:atk-bridge
WINDOWPATH=2
TERM=xterm-256color
VTE_VERSION=5202
SHELL=/bin/bash
QT_IM_MODULE=ibus
XMODIFIERS=@im=ibus
IM_CONFIG_PHASE=2
NVM_BIN=/home/kat/.nvm/versions/node/v6.14.4/bin
XDG_CURRENT_DESKTOP=ubuntu:GNOME
GPG_AGENT_INFO=/run/user/1001/gnupg/S.gpg-agent:0:1
GNOME_TERMINAL_SERVICE=:1.102
NVM_NODEJS_ORG_MIRROR=https://nodejs.org/dist
SHLVL=1
XDG_SEAT=seat0
MANPATH=/home/kat/.nvm/versions/node/v6.14.4/share/man:/usr/local/man:/usr/local/share/man:/usr/share/man:/usr/lib/jvm/java-8-oracle/man
GDMSESSION=ubuntu
GNOME_DESKTOP_SESSION_ID=this-is-deprecated
LOGNAME=kat
DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1001/bus
XDG_RUNTIME_DIR=/run/user/1001
XAUTHORITY=/run/user/1001/gdm/Xauthority
XDG_CONFIG_DIRS=/etc/xdg/xdg-ubuntu:/etc/xdg
PATH=/home/kat/.nvm/versions/node/v6.14.4/bin:/home/kat/.local/share/umake/bin:/home/kat/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/usr/lib/jvm/java-8-oracle/bin:/usr/lib/jvm/java-8-oracle/db/bin:/usr/lib/jvm/java-8-oracle/jre/bin:/opt/gradle/gradle-4.9/bin:/opt/packer:/opt/google/actions:/opt/ngrok
ANDROID_EMULATOR_USE_SYSTEM_LIBS=1
GJS_DEBUG_TOPICS=JS ERROR;JS LOG
SESSION_MANAGER=local/buck:@/tmp/.ICE-unix/2561,unix/buck:/tmp/.ICE-unix/2561
CAC_TRIPPLANNER_MEMORY=6144
LESSOPEN=| /usr/bin/lesspipe %s
GTK_IM_MODULE=ibus
OLDPWD=/projects/cac-tripplanner/otp_data
_=/usr/bin/env
</Details>

rbreslow commented:
I setup a test environment on Ubuntu 16.04. 

```bash
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.4 LTS
Release:        16.04
Codename:       xenial
```
I was not able to replicate a failure using the `fetch` module, but I encountered the `rsync` password prompt. I was able mitigate this by adding the `use_ssh_args` parameter to `synchronize`.

```bash
$ vagrant up --provision otp
...
TASK [cac-tripplanner.otp-data : Create Directory for OSM and GTFS] ************
ok: [otp] => {"changed": false, "gid": 0, "group": "root", "mode": "0775", "owner": "opentripplanner", "path": "/var/otp", "size": 4096, "state": "directory", "uid": 999}

TASK [cac-tripplanner.otp-data : Copy Feed Validator] **************************
ok: [otp] => {"changed": false, "checksum": "51200fe26444390159f7f1b83d7f32b16e2dad25", "dest": "/var/otp/validate_feed.py", "gid": 0, "group": "root", "mode": "0644", "owner": "root", "path": "/var/otp/validate_feed.py", "size": 1335, "state": "file", "uid": 0}

TASK [cac-tripplanner.otp-data : Install AuthBind (for serving OTP on port 80)] ***
ok: [otp] => {"cache_update_time": 1537891314, "cache_updated": false, "changed": false}

TASK [cac-tripplanner.otp-data : Touch AuthBind file if it does not exist] *****
changed: [otp] => {"changed": true, "dest": "/etc/authbind/byport/80", "gid": 0, "group": "root", "mode": "0755", "owner": "opentripplanner", "size": 0, "state": "file", "uid": 999}

TASK [cac-tripplanner.otp-data : Change owner and mode of byport file] *********
ok: [otp] => {"changed": false, "gid": 0, "group": "root", "mode": "0755", "owner": "opentripplanner", "path": "/etc/authbind/byport/80", "size": 0, "state": "file", "uid": 999}

TASK [cac-tripplanner.otp-data : Overwrite upstart service] ********************
changed: [otp] => {"changed": true, "checksum": "83157d474c50ab428a0311a5e8710624f1abcf1d", "dest": "/etc/init/otp.conf", "gid": 0, "group": "root", "md5sum": "6887db8f99889d56242f7983588cedcc", "mode": "0644", "owner": "root", "size": 559, "src": "/home/vagrant/.ansible/tmp/ansible-tmp-1537891339.59-79920457287044/source", "state": "file", "uid": 0}

TASK [cac-tripplanner.otp-data : Create data directory (test)] *****************
ok: [otp] => {"changed": false, "gid": 0, "group": "root", "mode": "0664", "owner": "root", "path": "./otp_data/", "size": 4096, "state": "directory", "uid": 0}

TASK [cac-tripplanner.otp-data : Download OTP Data (test)] *********************
skipping: [otp] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [cac-tripplanner.otp-data : Copy OTP Data (test/develop)] *****************
changed: [otp] => {"changed": true, "dest": "/var/otp/", "src": "/home/azavea/cac-tripplanner/deployment/ansible/roles/cac-tripplanner.otp-data/files/./otp_data/"}

TASK [cac-tripplanner.otp-data : Check for Existing Graph (develop)] ***********
ok: [otp] => {"changed": false, "stat": {"atime": 1537891347.1917198, "attr_flags": "e", "attributes": ["extents"], "block_size": 4096, "blocks": 499336, "charset": "binary", "checksum": "adaedf19a46d89d594cf8f5ea68f89b65ef5da16", "ctime": 1537891348.34372, "dev": 2049, "device_type": 0, "executable": false, "exists": true, "gid": 0, "gr_name": "root", "inode": 264038, "isblk": false, "ischr": false, "isdir": false, "isfifo": false, "isgid": false, "islnk": false, "isreg": true, "issock": false, "isuid": false, "mimetype": "application/octet-stream", "mode": "0664", "mtime": 1537891346.9477198, "nlink": 1, "path": "/var/otp/Graph.obj", "pw_name": "root", "readable": true, "rgrp": true, "roth": true, "rusr": true, "size": 255657303, "uid": 0, "version": "1506763921", "wgrp": true, "woth": false, "writeable": true, "wusr": true, "xgrp": false, "xoth": false, "xusr": false}}

TASK [cac-tripplanner.otp-data : Build OTP Graph (test/develop)] ***************
skipping: [otp] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [cac-tripplanner.otp-data : Copy Built OTP Graph to Host (develop)] *******
changed: [otp] => {"changed": true, "cmd": "/usr/bin/rsync --delay-updates -F --compress --archive --rsh=/usr/bin/ssh -S none -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o IdentityFile=/home/azavea/cac-tripplanner/.vagrant/machines/otp/virtualbox/private_key -o ControlMaster=auto -o ControlPersist=60s --rsync-path=sudo rsync --out-format=<<CHANGED>>%i %n%L vagrant@192.168.8.26:/var/otp/Graph.obj /home/azavea/cac-tripplanner/otp_data/", "msg": ">f..tp..... Graph.obj\n", "rc": 0, "stdout_lines": [">f..tp..... Graph.obj"]}

TASK [cac-tripplanner.otp-data : Create Graph Directory] ***********************
ok: [otp] => {"changed": false, "gid": 0, "group": "root", "mode": "0755", "owner": "root", "path": "/var/otp/default", "size": 4096, "state": "directory", "uid": 0}

TASK [cac-tripplanner.otp-data : Copy Router Configuration to Graph Directory] ***
ok: [otp] => {"changed": false, "checksum": "d361dc5b5135cce4679e8e0211c89c2190ccd492", "dest": "/var/otp/default/router-config.json", "gid": 0, "group": "root", "mode": "0644", "owner": "root", "path": "/var/otp/default/router-config.json", "size": 217, "state": "file", "uid": 0}

TASK [cac-tripplanner.otp-data : Move Graph to Graph Directory (test/develop)] ***
changed: [otp] => {"changed": true, "cmd": ["mv", "/var/otp/Graph.obj", "/var/otp/default"], "delta": "0:00:00.191813", "end": "2018-09-25 16:02:49.259785", "rc": 0, "start": "2018-09-25 16:02:49.067972", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [cac-tripplanner.otp-data : Copy Local Graph to Graph Directory (production)] ***
skipping: [otp] => {"changed": false, "skip_reason": "Conditional result was False"}

RUNNING HANDLER [azavea.opentripplanner : Restart OpenTripPlanner] *************
changed: [otp] => {"changed": true, "name": "otp", "state": "started"}

PLAY RECAP *********************************************************************
otp                        : ok=34   changed=9    unreachable=0    failed=0   
```

@flibbertigibbet Could you pull and try `vagrant up --provision otp` again?
